### PR TITLE
Add API tests Part 2

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## Description

Provide a summary of the modification as well as the issue that has been resolved.

- [ ] Need to add tests for validate and predict

## Changes

<details>
<summary>Describe the changes you made</summary>

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

File: src/anomalib/models/__init__.py [hunk 1]
```diff
@@ -74,61 +73,78 @@ def _convert_pascal_to_snake_case(pascal_case: str) -> str:
...
 
     Args:
-        config (DictConfig | ListConfig | str): Can either be a configuration or a string.
+        model (DictConfig | str): Can either be a configuration or a string.
+        *args: Variable length argument list for model init.
+        **kwdargs: Arbitrary keyword arguments for model init.
 
     Examples:
         >>> get_model("Padim")
+        >>> get_model("efficient_ad")
```
Comment (samet-akcay) [inline src/anomalib/models/__init__.py:144]: Nice!

File: src/anomalib/models/__init__.py [hunk 2]
```diff
@@ -74,61 +73,78 @@ def _convert_pascal_to_snake_case(pascal_case: str) -> str:
...
+    if model_class is None:
         logger.exception(f"Could not find the model {name}. Available models are {get_available_models()}")
         raise UnknownModelError
 
-    return model
+    return model_class
 
 
-def get_model(config: DictConfig | ListConfig | str | Namespace) -> AnomalyModule:
+def get_model(model: DictConfig | str | dict | Namespace, *args, **kwdargs) -> AnomalyModule:
```
Comment (samet-akcay) [inline src/anomalib/models/__init__.py:134]: Loved this function!

File: tests/integration/model/test_models.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+import pytest
+from lightning.pytorch.callbacks import ModelCheckpoint
+
+from anomalib.data import AnomalibDataModule, MVTec, TaskType, UCSDped
+from anomalib.engine import Engine
+from anomalib.models import AnomalyModule, get_available_models, get_model
+
+
+def models() -> list[str]:
```
Comment (samet-akcay) [inline tests/integration/model/test_models.py:20]: Would this be needed elsewhere? Maybe placed into a conftest as a fixture in the future?

File: tests/integration/model/test_models.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    ) -> tuple[AnomalyModule, AnomalibDataModule, Engine]:
+        """Return model, dataset, and engine objects.
+
+        Args:
+            model_name (str): Name of the model to train
+            dataset_path (Path): Path to the root of dummy dataset
+            project_path (Path): path to the temporary project folder
+
+        Returns:
+            tuple[AnomalyModule, AnomalibDataModule, Engine]: _description_
```
Comment (samet-akcay) [inline tests/integration/model/test_models.py:76]: `_description_` missing here.

**Review by samet-akcay — CHANGES_REQUESTED** [review]
Read very well! Just a single comment

File: tests/integration/model/test_models.py (hunk 3 above)
Comment (ashwinvaidya17) [reply tests/integration/model/test_models.py:20]: Once we fix `ai_vad` then we can get rid of this fixture and call `get_avaliable_models` here. Plus this is the only test that uses all the models. The others use only Padim or a random model

Comment (samet-akcay) [reply tests/integration/model/test_models.py:20]: ah right. Thanks for the clarification!

**Review by samet-akcay — APPROVED** [review]
looking good thanks!

**Pull request merged**